### PR TITLE
UMD wrap sdk.js

### PR DESCRIPTION
**Pull request by sankethkatta**
In order to allow the SDK to be installed and imported via npm as well as through inclusion as a script tag we wrapped the `sdk.js` code in the [Universal Module Definition](https://github.com/umdjs/umd)'s [`returnExports`](https://github.com/umdjs/umd/blob/master/templates/returnExports.js) template.

In addition, the gulpfile was refactored for readability, task naming, and logic flow. Now, the final output has 3 folders:
```
dist
  |____ cdn
    |____ sdk-2.0.0.js           // UMD wrapped, babel-ed, uglify-ed
    |____ redirect-2.0.0.js      // babel-ed, uglify-ed
    |____ redirect-2.0.0         // HTML file without extension
  |____ npm
    |____ sdk.js                 // UMD wrapped, babel-ed 
  |____ umd
    |____ sdk.js                 // UMD wrapped
  ```

+ `cdn`: is deployed to the javascript-sdk S3 bucket populating the CDN
+ `npm`: is ready for publishing to npm
+ `umd`: is a UMD wrapped version of `sdk.js` used for unit testing. this differs from the npm version as it is **not** babeled. we use this for unit testing so that coverage reports do not try to instrument code paths introduced by babel

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -5,57 +5,97 @@
...
+    .pipe(gulp.dest('dist/umd'));
+});
 
-// builds js files by babeling, uglifying, and versioning
-gulp.task('build-js', function() {
+/**
+ * Build sdk.js for npm publishing.
+ */
+gulp.task('build:npm', ['build:umd'], function() {
+  return gulp.src('src/sdk.js')
```
Comment (gurpreetatwal) [inline gulpfile.js]: shouldn't this be sourcing `dist/umd/sdk.js`?

File: gulpfile.js [hunk 2]
```diff
@@ -5,57 +5,97 @@
...
+    .pipe(babel())
+    .pipe(gulp.dest('dist/npm'));
+});
+
+/**
+ * Build JS for CDN publishing.
+ */
+gulp.task('build:cdn:js', ['build:umd'], function() {
   return gulp.src('src/*.js')
```
Comment (gurpreetatwal) [inline gulpfile.js]: same here

File: gulpfile.js [hunk 3]
```diff
@@ -5,57 +5,97 @@
...
 });
 
-// builds html file by templating and versioning
-gulp.task('build-html', function() {
+/**
+ * Build HTML for CDN publishing
+ */
+gulp.task('build:cdn:html', function() {
   return gulp.src('src/redirect.html')
     .pipe(template({redirectJS: `'/redirect-${version}.js'`}))
```
Comment (gurpreetatwal) [inline gulpfile.js]: nit: single quotes in HTML

File: gulpfile.js (hunk 1 above)
Comment (sankethkatta) [reply gulpfile.js]: yes

File: gulpfile.js (hunk 2 above)
Comment (sankethkatta) [reply gulpfile.js]: yes

File: gulpfile.js (hunk 3 above)
Comment (sankethkatta) [reply gulpfile.js]: :+1: i'm moving the string into the HTML file and only passing the version through the template dat

**Pull request merged**